src/GNDStk/v1.9/transport/CrossSection.hpp

```diff
@@ -149,6 +149,10 @@ class CrossSection : public Component<CrossSection> {
       XYs1d_regions1d(label) = obj; return *this;
    }
 
+   // XYs1d_regions1d(value) for vector push_back
+   CrossSection &XYs1d_regions1d(const XYs1d_regions1d_t &obj)
+      { setter(XYs1d_regions1d(), obj); return *this; }
```

Comment:
Since you and I last spoke in detail about the code that the code generator builds, I've made various tweaks to the getters and setters, based on my experience in using it for Tom's work.

Something that came up early was that doing things with generated classes proved to be more syntactically messy than I'd anticipated. One would need to write, for example, `()` to get an underlying vector, before doing a `push_back()`. With optionals, a `*` or `->` was needed, and those would fail if there was nothing in the optional.

I made various improvements so that the use of the generated classes would be much smoother and more natural. For example, imagine wanting to push a new value into an optional vector that currently had no value. (Meaning the optional had nothing yet, not that it had a vector with no elements.) Doing so would have been very confusing to users, so I wrote functions -- like the one you're looking at -- precisely to give users a simple way of doing things.

In terms of your concern about the requirements of GNDS, two remarks...

First, I'd note that our primary goal is of course to support GNDS. However, an important secondary goal is to allow other non-GNDS data structures to be created. With that secondary goal in mind, some of what I've done in the various open pull requests is geared towards saying, essentially, "let's make the code generator more general, so that it doesn't impose GNDS-isms on anyone." It's possible, for example, that someone *might* want to design classes that have non-unique label or index. Jane User's `label` or `index` might have a completely different meaning that their meaning in GNDS.

Second, the concern about unique GNDS `index` and `label` is certainly valid, but I think that the issue comes up anyway, no matter what we do. Whatever mechanism we provide for adding vector elements, somebody might try to add two elements that both have the same `index` or `label`, if those fields exist in the element type. What you're seeing here is a convenient shorthand, avoiding cluttered syntax, for adding an element. Guaranteeing the uniqueness you speak of is really a different issue.